lib/trino-parquet/src/main/java/io/trino/parquet/ParquetTypeUtils.java

```diff
@@ -129,27 +129,23 @@ private static Optional<ColumnDescriptor> getDescriptor(List<PrimitiveColumnIO>
 
     private static int getPathIndex(List<PrimitiveColumnIO> columns, List<String> path)
     {
-        int maxLevel = path.size();
```

Comment:
The original method was looking at prefix, hence `maxLevel`.
The new method compares changes that semantics because now there is `fieldPath.length != path.size()` check.